### ISSUE TITLE
Build fails on GHC 8.0.1 on OS X

### ISSUE THREAD

**Issue by kejace** [fumieval/extensible#6]
```
    /private/var/folders/yx/8tj3yfqs2qvgswckc2_xy70w0000gn/T/stack12408/extensible-0.3.7/src/Data/Extensible/Product.hs:125:1: warning: [-Woverlapping-patterns]
        Pattern match is redundant
        In an equation for ‘hzipWith’: hzipWith _ _ Nil = ...

    /private/var/folders/yx/8tj3yfqs2qvgswckc2_xy70w0000gn/T/stack12408/extensible-0.3.7/src/Data/Extensible/Product.hs:131:1: warning: [-Woverlapping-patterns]
        Pattern match is redundant
        In an equation for ‘hzipWith3’: hzipWith3 _ _ Nil _ = ...

    /private/var/folders/yx/8tj3yfqs2qvgswckc2_xy70w0000gn/T/stack12408/extensible-0.3.7/src/Data/Extensible/Product.hs:132:1: warning: [-Woverlapping-patterns]
        Pattern match is redundant
        In an equation for ‘hzipWith3’: hzipWith3 _ _ _ Nil = ...
    ghc: panic! (the 'impossible' happened)
      (GHC version 8.0.1 for x86_64-apple-darwin):
    	pprIfaceCo

    Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug
```

**Comment by kejace** [comment]
I will submit this to GHC later, but I thought this might be of interest here too.

**Comment by fumieval** [comment]
Actually, this bug is already fixed in 8.0.2: https://ghc.haskell.org/trac/ghc/ticket/12476

I guess we have to wait a bit more.

**Comment by fumieval** [comment]
Closing this since 8.0.2 is released.
